### DISCUSSION TITLE
Could we calculate the impact of Algorithm selection on overall carbon emission of the software?

### DISCUSSION THREAD

**Discussion by akshitbatra** [Green-Software-Foundation/sci#260] [Q&A]
I was wondering if it is possible to calculate the SCI score of different Algorithms available across the internet and observe their impact over the software carbon emissions.

**Comment by atg-abhishek** [comment]
Thanks for raising this question @akshitbatra - could you elaborate with a few examples of what you had in mind? Thanks!

**Reply by akshitbatra** [reply]
My idea is around as we know that currently for every algorithm we can find two factors i.e Time complexity and Space complexity of an algorithm. I was wondering if we can have one more factor like Carbon Complexity which we can calculate using SCI standard by assuming we run this algorithm as a software instance on a cloud and create a basic calculation. so whenever we use the algorithm we also consider the carbon complexity of an algorithm.

I would love to have more discussions around it, also would need some help and opinions on this.
